boards/arm/mimxrt1050_evk/Kconfig.defconfig

```diff
@@ -10,6 +10,15 @@ if BOARD_MIMXRT1050_EVK || BOARD_MIMXRT1050_EVK_QSPI
 config BOARD
 	default "mimxrt1050_evk"
 
+config NXP_IMX_RT_BOOT_HEADER
+	default y
+
+config DEVICE_CONFIGURATION_DATA
+	default y
+
+config NOCACHE_MEMORY
```

Comment:
`NOCACHE_MEMORY` is user configurable, so if a user disables it through menuconfig, the ethernet driver won't work. I think it should be better to change `drivers/ethernet/Kconfig.mcux` to:

```
menuconfig ETH_MCUX
        bool "MCUX Ethernet driver"
        depends on HAS_MCUX
        select NOCACHE_MEMORY if CPU_CORTEX_M7
        select ARM_MPU if CPU_CORTEX_M7
```

This way, it ensures the ethernet driver always work.